### PR TITLE
Theme/navigation refactor

### LINKED ISSUE #273: Reorganize and refactor Navigation
As discussed internally, we're going to break navigation components (mobile, desktop, etc...) out into separate partials.

### PR DESCRIPTION
**Pull request by jomurgel**
Closes #273 

### DESCRIPTION ###
- Removed the `_mobile-navigation.scss` partial which appeared to be a carry-over from the older mobile-navigataion setup.
- Created a `navigation` folder inside modules and moved navigation partials in there.
- Consolidated navigation into global (contains .menu), main, mobile (contains off-canvas) and post-navigation.
- Moved the post-navigation styles out of `_navigation.scss` into its own partial in `modules > navigation`.
- Moved the global link styles into it's own `_links.scss` partial because it's independent from any navigation (presumably) and added a transition helper.
- Removed the hamburger menu styles from `_icons.scss` both because it doesn't really belong in there and to consolidate all navigation styles into the new menus folder.
- Added support for multiple nested levels in the `mobile-menu.js`
- Added styles to rotate dropdown arrow at each level.
- Adds script to prevent body from scrolling when off-canvas navigation is open.
- Moved some of the basic container styles for the main navigation into the `_header.scss` partial to consolidate all of the structural elements.
- Fixed some positioning problems with the mobile hamburger on mobile if the admin bar was present.
- Reworked the transitions on the hamburger button to make them a little less choppy using transforms rather than absolute positioning.  Tested working still in IE11, Edge, Safari, Chrome, Firefox and Opera :grin: 
- Decreased the max transition duration from 0.8s to 0.5s in the transitions helper partial.  Adds transition for hamburger specifically.
- Reworked the header/nav area on desktop to fix some basic render issues.
- Removed as much fat, or consolidated as much as I could to clean things up a bit.
- Added table of contents to `_mobile-navigation.scss` partial for ease of use.

### SCREENSHOTS ###
**Before**
kapture 2017-10-03 at 8 16 47

**After**
kapture 2017-10-03 at 9 58 56

**Before**
screenshot 2017-10-03 09 16 52

**After**
screenshot 2017-10-03 11 53 16

### OTHER ###
- [x] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [x] Does this pass CBT?

### STEPS TO VERIFY ###
Use branch and test mobile/desktop navigation.

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?
I don't believe so.

### CODE REVIEW

File: assets/sass/modules/_header.scss [hunk 1]
```diff
@@ -44,18 +21,60 @@
 			width: 50%;
 		}
 	} // img
+
+	// Site Title
```
Comment (gregrickaby) [inline assets/sass/modules/_header.scss]: @jomurgel Can you use the WDS standard, multiple line comment style throughout this file?

```
//-----------------------------------------
// I'm a selector comment block 
//-----------------------------------------
```

File: assets/sass/modules/navigation/_global-navigation.scss [hunk 2]
```diff
@@ -0,0 +1,60 @@
+//--------------------------------------------------------------
+// Global and Shared Navigation Styles
+//--------------------------------------------------------------
+
+// Default menu (active after a fresh WordPress install)
```
Comment (gregrickaby) [inline assets/sass/modules/navigation/_global-navigation.scss]: @jomurgel All single line comments should end with punctuation. I know it's inconsistent throughout this theme, but if you could fix up the comments in this PR, that'd be 👍

**Review by gregrickaby — CHANGES_REQUESTED** [review]
Just a couple of quick comment tweaks! Thanks!

**Comment by jomurgel** [comment]
@gregrickaby I've broken out the `_header.scss` blocks and added block documentation as requested.  I've also ensured that all single-line (or otherwise) comments in files for this PR terminate in a period. 👍

**Review by gregrickaby — APPROVED** [review]

**Pull request merged**